pkg/storegateway/indexheader/lazy_binary_reader_test.go

```diff
@@ -383,3 +383,36 @@ func TestLazyBinaryReader_ConcurrentLoadingOfSameIndexReader(t *testing.T) {
 		require.Fail(t, "goroutines did not finish in time")
 	}
 }
+
+func TestLazyBinaryReader_SymbolReaderAndUnload(t *testing.T) {
+	tmpDir, bkt, blockID := initBucketAndBlocksForTest(t)
+
```

Comment:
```suggestion
func TestLazyBinaryReader_SymbolReaderAndUnload(t *testing.T) {
  t.Parallel()
	tmpDir, bkt, blockID := initBucketAndBlocksForTest(t)

```
nit: you can add a `t.Parallel()` so we don't slow down the whole package's tests with the `time.After(time.Second)`

Reply:
Done. There aren't many other parallel tests in the package (except `TestStreamBinaryReader_CheckSparseHeadersCorrectnessExtensive`), so the effect is limited at the moment.